src/bcccp/carpark/paystation/PaystationController.java

```diff
@@ -6,15 +6,16 @@
 public class PaystationController 
 		implements IPaystationController {
 	
-	enum State
-    {
-		IDLE, WAITING, REJECTED, PAID
-    }
-	private IPaystationUI ui;	
-	private ICarpark carpark;
-	protected State state;
-	private IAdhocTicket  adhocTicket = null;
-	private float charge;
+private enum STATE { IDLE, WAITING, REJECTED, PAID } 
```

Comment:
private enum STATE declaration good choice.

Reply:
yes it is. did you try to execute this code?